### ISSUE TITLE
requires package-lint 0.2?

### ISSUE THREAD

**Issue by progfolio** [purcell/flycheck-package#11]
https://github.com/purcell/flycheck-package/blob/3a6aaed29ff61418c48c0251e1432c30748ae739/flycheck-package.el#L10

Is there a package-lint 0.2?
It looks like package-lint declares a Version of "0":

https://github.com/purcell/package-lint/blob/8f7c46d204c9ebe99e3843934be726fa3a3e5190/package-lint.el#L9

For some context as to why I'm asking:

straight.el and Elpaca have been asked to issue warnings when a package requires a higher version of a dependency than that which is installed. While developing the feature, flycheck-package requesting package-lint 0.2 is the only package which was flagged (on my personal init).

**Comment by purcell** [comment]
Yes, you can see the package here: https://stable.melpa.org/#/package-lint (0.17, in fact!)

The MELPA Stable build machinery fills in the headers using the tag. It would be invalid for straight/elpaca to assume that the headers in the files in a git repository would always be correct: many repos use placeholder versions like this and rely on the packaging process to fill them in.

**Comment by progfolio** [comment]
> many repos use placeholder versions like this and rely on the packaging process to fill them in.

Do we have any numbers on this?
I think it's actually a pretty rare case.
Most package I encounter maintain the Version metadata in the source file despite a *-pkg.el file being generated by an ELPA's build system.

https://github.com/search?q=%22%3B%3B+Version%3A%22++language%3Aelisp+NOT+is%3Afork&type=code&p=5

**Comment by purcell** [comment]
No, I don't have stats, but I have about 57 of 'em just directly under my user account here, several quite popular, so that's "many", and I've seen it elsewhere too. I know there's a push by the GNU folks to use `;; Version` headers in the source code, but it means constant bumping of that for each release, and it has consequences for dev versions and branching schemes. How would `straight` or `elpaca` even find stable releases in git repos? It seems those tools would need to be told to use tags, or look in certain branches anyway.

**Comment by progfolio** [comment]
> How would straight or elpaca even find stable releases in git repos? It seems those tools would need to be told to use tags, or look in certain branches anyway.

That's not the goal of the feature. The goal is to warn users when a package declares a dependency version greater than the version the user has installed. For more context:

https://github.com/radian-software/straight.el/issues/1083
